webpack/InsightsCloudSync/Components/InsightsTable/InsightsTableActions.js

```diff
@@ -117,13 +131,29 @@ export const onTableSelect = (
     });
 
     showSelectAllAlert = true;
+  } else if (isSelected) {
+    selectedIds[results[rowId].id] = true;
+    if (results.length === Object.keys(selectedIds).length) {
+      showSelectAllAlert = true;
+    }
   } else {
-    isSelected
-      ? (selectedIds[results[rowId].id] = true)
-      : delete selectedIds[results[rowId].id];
+    dispatch(setSelectAll(false));
+    delete selectedIds[results[rowId].id];
```

Comment:
did you mean to put it inside the following `else` on line 139?

Reply:
What do you mean?

Reply:
I mean that in line `126` we check if the main checkbox was selected and in line `139` else we refer to all other rows checkboxes. in each of the if's context, you can do whatever you want, but do you really need the new `else if`?

Reply:
The new else if replaces the condition that was in the else so
``` javascript
else {
   isSelected? dispatch(setSelectAll(false)) : (selectedIds[results[rowId].id] = true)
}
```
is now 
``` javascript
else if(isSelected) {
   dispatch(setSelectAll(false))
   ...
} else {
   (selectedIds[results[rowId].id] = true)
   ...
}
```

Reply:
Moved the else code to a function for readability